prediction_market_agent/agents/prophet_agent/deploy.py

```diff
@@ -14,22 +13,73 @@
 from prediction_market_agent_tooling.tools.betting_strategies.stretch_bet_between import (
     stretch_bet_between,
 )
+from prediction_market_agent_tooling.tools.langfuse_ import observe
 from prediction_market_agent_tooling.tools.utils import (
     prob_uncertainty,
     should_not_happen,
 )
 from prediction_prophet.benchmark.agents import (
     EmbeddingModel,
     OlasAgent,
+    Prediction,
     PredictionProphetAgent,
+    Research,
 )
 
 from prediction_market_agent.utils import DEFAULT_OPENAI_MODEL
 
 
+class PredictionProphetAgentObserved(PredictionProphetAgent):
+    @observe()
+    def research(
+        self, market_question: str, add_langfuse_callback: bool = False
+    ) -> Research:
+        return super().research(
+            market_question=market_question,
+            add_langfuse_callback=add_langfuse_callback,
+        )
+
+    @observe()
+    def make_prediction(
+        self,
+        market_question: str,
+        additional_information: str,
+        add_langfuse_callback: bool = False,
+    ) -> Prediction:
+        return super().make_prediction(
+            market_question=market_question,
+            additional_information=additional_information,
+            add_langfuse_callback=add_langfuse_callback,
+        )
+
+
+class OlasAgentObserved(OlasAgent):
+    @observe()
+    def research(
+        self, market_question: str, add_langfuse_callback: bool = False
+    ) -> str:
+        return super().research(
+            market_question=market_question,
+            add_langfuse_callback=add_langfuse_callback,
+        )
```

Comment:
Having these extra subclasses just for Langfuse could be avoided (see https://github.com/gnosis/prediction-market-agent-tooling/pull/345/files#r1718609063). But also OK if done like this.

Reply:
Yes, removed.